python/py-allennlp/Portfile

```diff
@@ -0,0 +1,123 @@
+                    port:py${python.version}-scipy \
+                    port:py${python.version}-spaCy \
+                    port:py${python.version}-sqlparse \
+                    port:py${python.version}-tensorboardX \
+                    port:py${python.version}-transformers \
+                    port:py${python.version}-tqdm \
+                    port:py${python.version}-tz \
+                    port:py${python.version}-unidecode \
+                    port:py${python.version}-word2number
+
+    depends_test-append \
+                    port:py${python.version}-flaky \
+                    port:py${python.version}-pytest \
+                    port:py${python.version}-responses
+
+    # Fix allennlp's dependecies on deprecated names pytorch-pretrained-bert|pytorch-transformers
+    # remove this after the upstream has been fixed
+    post-extract {
+        # pytorch-pretrained-bert|pytorch-transformers
+        reinplace -E \
+            {/^[[:space:]]*(pytorch-pretrained-bert|pytorch-transformers)([[:space:]]*(<|>|=)=?[[:space:]]*[[:digit:]]+(\.[[:digit:]]+)*)?[[:space:]]*/d} \
+            ${worksrcpath}/requirements.txt
+        reinplace -E \
+            {/^[[:space:]]*['"](pytorch-pretrained-bert|pytorch-transformers)([[:space:]]*(<|>|=)=?[[:space:]]*[[:digit:]]+(\.[[:digit:]]+)*)?['"],[[:space:]]*$/d} \
+            ${worksrcpath}/setup.py
+        fs-traverse f ${worksrcpath} {
+            if {[file isfile ${f}]
+                && [string match "*.py" ${f}]} {
+                reinplace -E -q \
+                    {s/((import|from)[[:space:]]+)(pytorch_pretrained_bert|pytorch_transformers)(\.[[:alnum:]_]+)*/\1transformers/g} \
+                    ${f}
+                reinplace -q \
+                    {s/BertAdam/AdamW/g} \
+                    ${f}
+            }
+        }
+        # conllu|overrides
+        reinplace -E \
+            {s/^([[:space:]]*(conllu|overrides)[[:space:]]*)==([[:space:]]*[[:digit:]]+(\.[[:digit:]]+)*)?/\1>=\3/g} \
+            ${worksrcpath}/requirements.txt
+        reinplace -E \
+            {s/^([[:space:]]*['"](conllu|overrides)[[:space:]]*)==([[:space:]]*[[:digit:]]+(\.[[:digit:]]+)*['"],)?/\1>=\3/g} \
+            ${worksrcpath}/setup.py
+    }
+
+    post-destroot {
+        set docdir ${prefix}/share/doc/${subport}
+        set tutorialsdir ${prefix}/share/${subport}/tutorials
+        xinstall -d \
+            ${destroot}${docdir} \
+            ${destroot}${tutorialsdir}
+        xinstall -m 0644 -W ${worksrcpath} README.md MODELS.md LICENSE \
+            ${destroot}${docdir}
+        copy ${worksrcpath}/tutorials/getting_started \
+            ${destroot}${tutorialsdir}
+    }
+
+    test.run        yes
+    test.env        PYTHONPATH=${worksrcpath}/build/lib
+    test.cmd        ${prefix}/bin/python${python.branch}
+    test.pre_args   {-c 'from allennlp.commands.test_install import TestInstall; TestInstall()'}
+    test.target
+
```

Comment:
shouldn't you just be able to run this with ```py.test-${python.branch}``` instead, that would be preferred I think.

Reply:
Yes, but `botocore` is behind on its dependencies too. See https://github.com/boto/botocore/issues/1872

This hackiness hacks around this issue, which causes a dependency failure.

We can unhack this when `botocore` gets around to fixing its dependency issues.

Reply:
I cannot quickly see how these two are related... But as long as this way testing actually works it’s fine with me.